components/compass-runtime-agent/internal/compass/director/client_test.go

```diff
@@ -271,107 +288,196 @@ func TestConfigClient_FetchConfiguration(t *testing.T) {
+				ExpectedReq:        expectedGetLabelsRequest,
+			})
 
 		configClient := NewConfigurationClient(gqlClient, runtimeConfig)
 
 		// when
-		labels, err := configClient.SetURLsLabels(runtimeURLsConfig)
+		reconciledLabels, err := configClient.ReconcileLabels(runtimeURLsConfig)
 
 		// then
-		require.Error(t, err)
-		assert.Nil(t, labels)
+		require.NoError(t, err)
+		assert.Equal(t, 0, len(reconciledLabels))
+	})
+
+	t.Run("should override URLs if there are already set but are different", func(t *testing.T) {
+		labelsResponse := graphql.Labels{}
+		labelsResponse[eventsURLLabelKey] = runtimeURLsConfig.EventsURL + " something different"
+		labelsResponse[consoleURLLabelKey] = runtimeURLsConfig.ConsoleURL + " something different"
+
+		gqlClient := gql.NewQueryAssertClient(t, false,
+			gql.ResponseMock{
+				ModifyResponseFunc: newGetExpectedLabelsFunc(labelsResponse),
+				ExpectedReq:        expectedGetLabelsRequest,
+			}, gql.ResponseMock{
+				ModifyResponseFunc: newSetExpectedLabelFunc(eventsURLLabel),
+				ExpectedReq:        expectedSetEventsURLRequest,
+			}, gql.ResponseMock{
+				ModifyResponseFunc: newSetExpectedLabelFunc(consoleURLLabel),
+				ExpectedReq:        expectedSetConsoleURLRequest,
+			})
+
+		configClient := NewConfigurationClient(gqlClient, runtimeConfig)
+
+		// when
+		reconciledLabels, err := configClient.ReconcileLabels(runtimeURLsConfig)
+
+		// then
+		require.NoError(t, err)
+		assert.Equal(t, 2, len(reconciledLabels))
+		assert.Equal(t, runtimeURLsConfig.EventsURL, reconciledLabels[eventsURLLabelKey])
+		assert.Equal(t, runtimeURLsConfig.ConsoleURL, reconciledLabels[consoleURLLabelKey])
 	})
 
-	t.Run("should return error if setting Console URL as label returned nil response", func(t *testing.T) {
-		expectedResponses := []*graphql.Label{nil, nil}
+	t.Run("should set only missing URLs as labels", func(t *testing.T) {
+		labelsResponse := graphql.Labels{}
+		labelsResponse[eventsURLLabelKey] = runtimeURLsConfig.EventsURL
 
-		gqlClient := gql.NewQueryAssertClient(t, false, newSetExpectedLabelFunc(expectedResponses), expectedSetEventsURLRequest, expectedSetConsoleURLRequest)
+		gqlClient := gql.NewQueryAssertClient(t, false,
+			gql.ResponseMock{
+				ModifyResponseFunc: newGetExpectedLabelsFunc(labelsResponse),
+				ExpectedReq:        expectedGetLabelsRequest,
+			},
+			gql.ResponseMock{
+				ModifyResponseFunc: newSetExpectedLabelFunc(consoleURLLabel),
+				ExpectedReq:        expectedSetConsoleURLRequest,
+			})
 
 		configClient := NewConfigurationClient(gqlClient, runtimeConfig)
 
 		// when
-		labels, err := configClient.SetURLsLabels(runtimeURLsConfig)
+		reconciledLabels, err := configClient.ReconcileLabels(runtimeURLsConfig)
+
+		// then
+		require.NoError(t, err)
+		assert.Equal(t, 1, len(reconciledLabels))
+		assert.Equal(t, runtimeURLsConfig.ConsoleURL, reconciledLabels[consoleURLLabelKey])
+	})
+
+	t.Run("should return error if getting labels returns nil response", func(t *testing.T) {
```

Comment:
This really sounds like a cool fault-tolerance improvement. 👍🏼 But I was wondering if it will help much though. Because if the `getLabels` call fails due to a Director outage or something, maybe the `setLabel` call will also fail. What do you think?

Reply:
In my opinion the closer we are to the old behaviour, the safer the new implementation is.

Reply:
I totally agree 👍🏼 but I think that this change here is small and basic enough to not be a concern. If you ask me, It's okay to fail if getting labels doesn't succeed, because if that happens the setting of labels will also fail.